internal/vds/vds.cpp

```diff
@@ -506,28 +427,31 @@ struct response fetch_fence(
         }
     };
 
+    const Axis inline_axis(axis_name::INLINE, layout);
+    const Axis crossline_axis(axis_name::CROSSLINE, layout);
+
     for (size_t i = 0; i < npoints; i++) {
         const float x = *(coordinates++);
         const float y = *(coordinates++);
 
         auto coordinate = transform_coordinate(x, y);
 
-        auto validate_boundary = [&] (const int voxel) {
+        auto validate_boundary = [&] (const Axis& axis) {
             const auto min = -0.5;
-            const auto max = layout->GetDimensionNumSamples(voxel_todim(voxel))
-                            - 0.5;
-            if(coordinate[voxel] < min || coordinate[voxel] >= max) {
+            const auto max = axis.get_number_of_points() - 0.5;
+            const int index = axis.get_index();
```

Comment:
Don't understand why are you using here something that belongs to axis.

voxel name originally is probably wrong (may be changed in separate commit?), but it is not related to axis itself.
 
It is related to interface. If we one day decide that coordinates should be provided in form 
`[cat_gif, dog_gif, hamster_gif, inline_coord, xline_coord]` then this thing would have to refer to coordinate[3] and coordinate[4] to get the data. Unrelated to axis.

Reply:
Agree with the renaming.

Regarding the index I can understand the concern. Does it also refer to the use of 
`axis.get_number_of_points()`? At least GitHub highlights both. At least it would not be obvious to me who/what else should know about the number of points/samples in a certain direction.